### DISCUSSION TITLE
Does `console.log` have any form of log levels?

### DISCUSSION THREAD

**Discussion by rlaphoenix** [Textualize/rich#2833] [Q&A] [answered]
It seems to me as if `console.log` only logs at INFO level, or rather NOTSET level. Is it possible to set the log level? I'm aware you can use logging with RichHandler to then do log.error and such, but I see no reason to use both logging and console for the same behavior. I've also noticed less terminal conflicts if I stick to one or the other, rather than using a mix of both.

I can inherit Console, override log, and super() call and override `style` with e.g., `style="logging.level.error"` but is this really the best solution?

Am I to assume we are still meant to use `logging` with a `RichHandler` for typical logs? What is the purpose of console.log then?

**Comment by willmcgugan** [comment, marked as answer]
`console.log` is a debug aid for those who like the `print` style of debugging. `console.log` will pretty print data and renderables. You don't typically leave it in your app.

Logging has an entirely different use case. Logging stays in your app. Logging works with strings only.  You can configure logging to send logs to different, or even multiple, locations. You would typically configure logging to use RichHandler for debugging, and somewhere else in production.

Two different use cases, even if they render in a similar way.
